database/sessions.ts

```diff
@@ -0,0 +1,56 @@
+import { cache } from 'react';
+import type { User } from '../migrations/00002-createTableUsers';
+import type { Session } from '../migrations/00004-createTableSessions';
+import { sql } from './connect';
+
+export const getValidSession = cache(async (sessionToken: Session['token']) => {
+  const [session] = await sql<Pick<Session, 'id' | 'token'>[]>`
+    SELECT
+      sessions.id,
+      sessions.token
+    FROM
+      sessions
+    WHERE
+      sessions.token = ${sessionToken}
+      AND expiry_timestamp > now()
+  `;
+  return session;
+});
+
+export const createSessionInsecure = cache(
+  async (token: Session['token'], userId: User['id']) => {
+    const [session] = await sql<Session[]>`
+      INSERT INTO
+        sessions (token, user_id)
+      VALUES
+        (
+          ${token},
+          ${userId}
+        )
+      RETURNING
+        sessions.id,
+        sessions.token,
+        sessions.user_id
+    `;
+
+    await sql`
+      DELETE FROM sessions
+      WHERE
+        expiry_timestamp < now()
+    `;
+
+    return session;
+  },
+);
```

Comment:
Let's follow our usual pattern of "Insecure" functions at the bottom of the file

(for all files)

Reply:
3680e991d5ebe02c7ac29e97ca90d7c6ec300c61